torch/distributed/elastic/rendezvous/c10d_rendezvous_backend.py

```diff
@@ -166,6 +167,23 @@ def _create_tcp_store(params: RendezvousParameters) -> TCPStore:
 
     return store
 
+def _create_file_store(params: RendezvousParameters) -> FileStore:
+    # The directory is readable, writable, and searchable only by the
+    # creating user ID. Same goes for the created file.
+    # TODO: Right now, this temporary directory and file does not clean itself
+    # up after use. In the future, a close method for rendezvous backend should
+    # be implemented to enable cleaning up resources. (For now, since it
+    # writes to /tmp, it is ok)
+    temp_dir = tempfile.mkdtemp(prefix="torch_rdzv_")
+    _, path = tempfile.mkstemp(dir=temp_dir, suffix=str(params.run_id))
+
+    # Min and max number of nodes should be the same so we can ensure
+    # num_workers, which is passed in to create the FileStore is correct.
+    if (params.min_nodes != params.max_nodes):
```

Comment:
This defeats the whole purpose of dynamic rendezvous handling. I encourage you to lift this limitation from `FileStore`. I just checked its source code and apparently the reason why we require `numWorkers` is to delete the file once all workers are complete (see [here](https://github.com/pytorch/pytorch/blob/master/torch/lib/c10d/FileStore.cpp#L286)).

Since we are already using a temporary file, there is really no need for us to explicitly delete it. Ideally we should make the `numWorkers` parameter of `FileStore` optional and skip the file deletion in the constructor if it is not set.